### PR TITLE
refactor gamma calculation

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- add a function `gamma<>()` to calculate the gamma of a particle
- add a forward declaration for the class `Gamma<>`

This change is needed to add filter functors for radiation in a follow up.

### CODE REVIEW

File: src/picongpu/include/algorithms/Gamma.hpp [hunk 1]
```diff
@@ -19,29 +19,36 @@
...
+        T_MassType const mass
+    ) const
     {
-        const valueType fMom2 = math::abs2( precisionCast<valueType >( mom ) );
-        const valueType c2 = SPEED_OF_LIGHT*SPEED_OF_LIGHT;
+        using namespace PMacc;
 
-        const valueType m2_c2_reci = valueType(1.0) / precisionCast<valueType >( mass * mass * c2 );
+        valueType const fMom2 = math::abs2( precisionCast< valueType >( mom ) );
+        valueType const c2 = SPEED_OF_LIGHT * SPEED_OF_LIGHT;
```
Comment (ax3l) [inline src/picongpu/include/algorithms/Gamma.hpp]: constexpr

File: src/picongpu/include/algorithms/Gamma.def [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+#include "simulation_defines.hpp"
+
+
+namespace picongpu
+{
+
+    /** calculate the gamma of a particle
+     *
+     * @tparam T_PrecisionType type in which the calculation is performed
```
Comment (ax3l) [inline src/picongpu/include/algorithms/Gamma.def]: type -> precision

File: src/picongpu/include/algorithms/Gamma.def [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        operator()(
+            T_MomType const & mom,
+            T_MassType const mass
+        ) const;
+
+    };
+
+    /** calculate the gamma of a particle
+     *
+     * @tparam T_PrecisionType type in which the calculation is performed
```
Comment (ax3l) [inline src/picongpu/include/algorithms/Gamma.def]: type -> precision

**Comment by psychocoderHPC** [comment]
changed

**Pull request merged**